lib/_stream_writable.js

```diff
@@ -285,9 +285,15 @@ Writable.prototype.write = function (chunk, encoding, cb) {
   var state = this._writableState;
   var ret = false;
 
-  var isBuf = !state.objectMode && _isUint8Array(chunk);
```

Comment:
Streams not operating under `objectMode` deal solely with Buffers, UInt8Arrays and Strings. IMO this check (`!state.objectMode`) is necessary and should be valid for all cases.

Reply:
Shouldn't `isBuf` be renamed to `isNotBuf` since `chunk` is a UInt8Array?